loopsPartOne.js

```diff
@@ -102,16 +109,32 @@ function doAction(action) {
    -------START OF YOUR CODE----------------------------------------------------
 */
 
-function performRoutine(routine) {
-    // Your Code Here!
+function performRoutine() {
+    while (exercise < actionsPerformed.index) {
```

Comment:
There's something wrong with this line.  exercise isn't defined yet, it's defined down on 116.  But it's an object, not a number.  
  
But you actually don't need this line.  The code after this solves the problem.